fe2-android/app/src/main/java/com/github/dedis/student20_pop/utility/handler/MessageHandler.java

```diff
@@ -0,0 +1,64 @@
+package com.github.dedis.student20_pop.utility.handler;
+
+import android.util.Log;
+import com.github.dedis.student20_pop.model.data.LAORepository;
+import com.github.dedis.student20_pop.model.data.LAOState;
+import com.github.dedis.student20_pop.model.network.method.message.MessageGeneral;
+import com.github.dedis.student20_pop.model.network.method.message.data.Data;
+import com.github.dedis.student20_pop.model.network.method.message.data.Objects;
+import com.github.dedis.student20_pop.model.network.method.message.data.lao.CreateLao;
+import com.github.dedis.student20_pop.model.network.method.message.data.lao.StateLao;
+import com.github.dedis.student20_pop.model.network.method.message.data.message.WitnessMessageSignature;
+
+/** General message handler class */
+public class MessageHandler {
+
+  public static final String TAG = MessageHandler.class.getSimpleName();
+
+  private MessageHandler() {
+    throw new IllegalStateException("Utility class");
+  }
+
+  /**
+   * Send messages to the corresponding handler.
+   *
+   * @param laoRepository the repository to access the messages and LAOs
+   * @param channel the channel on which the message was received
+   * @param message the message that was received
+   * @return true if the message cannot be processed and false otherwise
```

Comment:
This is a very atypical return value.
In general `true` signifies success, and `false` failure.
If possible, I'd switch it the other way around.

Reply:
The return value represents if the message is queued or not, I can swap it to represent success instead